modules/payloads/singles/windows/meterpreter_bind_named_pipe.rb

```diff
@@ -40,6 +40,9 @@ def generate(opts={})
 
   def generate_config(opts={})
     opts[:uuid] ||= generate_payload_uuid
+    opts[:debug_build] ||= datastore['MeterpreterDebugBuild']
+    parsed_options = Msf::OptMeterpreterDebugLogging.parse_logging_options(opts[:debug_build])
```

Comment:
I thinks some confusion of variables happened here.
```suggestion
    parsed_options = Msf::OptMeterpreterDebugLogging.parse_logging_options(datastore['MeterpreterDebugLogging'])
```

I wonder how putting a helper in msf/base/sessions/meterpreter_options.rb that grabs the logging options might work it looks like that `mixin` is already in these locations:

Something like adding to MeterpreterOptions:
```Ruby
      def meterpreter_logging_config(opts = {})
        ds = opts[:datastore] || datastore
        {
          debug_build: (ds[:debug_build] || datastore['MeterpreterDebugBuild']),
          log_path:    (ds[:log_path] || Msf::OptMeterpreterDebugLogging.parse_logging_options(datastore['MeterpreterDebugLogging'])[:rpath])
        }
      end
```
and then
```Ruby
   config_opts = {
      arch:       opts[:uuid].arch,
      exitfunk:   datastore['EXITFUNC'],
      expiration: datastore['SessionExpirationTimeout'].to_i,
      uuid:       opts[:uuid],
      transports: [transport_config_reverse_ipv6_tcp(opts)],
      extensions: (datastore['EXTENSIONS'] || '').split(','),
      ext_init:   (datastore['EXTINIT'] || ''),
      stageless:  true
    }.merge(meterpreter_logging_config(opts))
```
similar to the pattern used for transport configs

Reply:
>I thinks some confusion of variables happened here.
Oh yep bad copy paste nice catch